sources/bioformats/large_image_source_bioformats/__init__.py

```diff
@@ -638,6 +638,7 @@ def getTile(self, x, y, z, pilImageAllowed=False, numpyAllowed=False, **kwargs):
             retile[0:min(tile.shape[0], finalHeight), 0:min(tile.shape[1], finalWidth)] = tile[
                 0:min(tile.shape[0], finalHeight), 0:min(tile.shape[1], finalWidth)]
             tile = retile
+        self._dtype = tile.dtype
```

Comment:
Rather than modifying any tile source's `getTile`, we could add code to tilesource/base.py's `_outputTile` method that, before applying a style, would do something like
```
if self._dtype is None:
    if tileEncoding == TILE_FORMAT_NUMPY:
        self._dtype = tile.dtype
    elif tileEncoding == TILE_FORMAT_PIL:
        self._dtype = numpy.uint8 if ';16' not in tile.mode else numpy.uint16
    else:
        self._dtype = _imageToNumpy(tile)[0].dtype
```

Reply:
Changed in [4f41f62](https://github.com/girder/large_image/pull/1144/commits/4f41f620606f4dc3408a1102341eec65d3ec3e41)